framework/include/utils/MooseTypes.h

```diff
@@ -27,16 +27,11 @@
 
 #include <string>
 #include <vector>
+#include <memory>
 
-#ifdef LIBMESH_HAVE_CXX11_SHARED_PTR
-#  include <memory>
-#  define MooseSharedPointer std::shared_ptr
-#  define MooseSharedNamespace std
-#else
-#  include "boost/shared_ptr.hpp"
-#  define MooseSharedPointer boost::shared_ptr
-#  define MooseSharedNamespace boost
-#endif
+// DO NOT USE (Deprecated)
+#define MooseSharedPointer std::shared_ptr
```

Comment:
Not with a `#define`. It should be possible to promote this define to an actual shim class and then we could do that. MooseSharedPointer is still sprinkled throughout the modules a lot and also in applications so we probably don't want to do that just yet. However we'll have to be diligent with watching these sneak back into the framework while we are transitioning. It's a #define so there really isn't a problem if they do crop up from time to time.